src/nav-bar/NavBar.js

```diff
@@ -46,54 +52,39 @@ const addLinkClasses = ({ isCurrent }) => ({
   }),
 });
 
-export default function NavBar({ large, onClick }) {
+export default function NavBar({ pages, onClick, navigationStyles, nested }) {
   return (
     <NavContainer>
-      <NavList large={large}>
-        <NavItem>
-          <Link getProps={addLinkClasses} to={PATHS.overview} onClick={onClick}>
-            Overview
-          </Link>
-        </NavItem>
-        <NavItem>
-          <Link
-            getProps={addLinkClasses}
-            to={PATHS.sentencing}
-            onClick={onClick}
-          >
-            Sentencing
-          </Link>
-        </NavItem>
-        <NavItem>
-          <Link getProps={addLinkClasses} to={PATHS.prison} onClick={onClick}>
-            Prison
-          </Link>
-        </NavItem>
-        <NavItem>
-          <Link
-            getProps={addLinkClasses}
-            to={PATHS.probation}
-            onClick={onClick}
-          >
-            Probation
-          </Link>
-        </NavItem>
-        <NavItem>
-          <Link getProps={addLinkClasses} to={PATHS.parole} onClick={onClick}>
-            Parole
-          </Link>
-        </NavItem>
+      <NavList navigationStyles={navigationStyles}>
+        {Array.from(pages, ([path, label]) => (
+          <NavItem key={path}>
+            <Link
+              getProps={addLinkClasses}
+              to={`${nested ? "../" : ""}${PATHS[path]}`}
+              onClick={onClick}
+            >
+              {label}
+            </Link>
+          </NavItem>
+        ))}
       </NavList>
     </NavContainer>
   );
 }
 
 NavBar.propTypes = {
-  large: PropTypes.bool,
+  pages: PropTypes.instanceOf(Map),
   onClick: PropTypes.func,
+  navigationStyles: PropTypes.shape({
+    ul: PropTypes.array,
+    li: PropTypes.array,
+  }),
```

Comment:
I'm not crazy about the way I ended up passing around styles to support the various visual treatments, but didn't really know how to implement a better approach.  The gist of it is that I was trying to limit the amount of code and style duplication while also having things look the way we want them to in these six scenarios:

- Sidebar Navigation (Desktop only - hidden on mobile)
- Branding Bar Navigation (Mobile only - hidden on desktop)
- Overview Page Navigation (Desktop and Mobile)
- Secondary Navigation (Desktop and Mobile)

Reply:
the only alternative I can really think of would just be to let parents of `NavBar` set a className (or pass some prop that maps to a classname) and then consolidate all the styles in this module. 

I guess it's arguable whether that's really any better, but it would at least keep all the CSS for this component in one place, which might make future changes easier since we wouldn't have to hunt down all the places that are passing in style overrides when we are debugging CSS issues. Don't feel super strongly about this though if you don't agree

Reply:
Yeah, that's what I originally tried, but was struggling to get everything wired such that the styles got overridden as necessary.  I think I just figured it out though.  Will add a new commit after our 2pm meeting